File: 김진호/2주차/2231.py
import sys

N = int(input())

res = sys.maxsize

for constructor_num in range(N):
    tmp = list(str(constructor_num))
    sum_ = constructor_num
    for num in tmp:
        sum_ += int(num)
    if sum_ == N and constructor_num < res:
        res = constructor_num

if res == sys.maxsize:
    print(0)
else:
    print(res)

File: 전효정/2주차/2231.c
N=int(input())

def Sep(i):
    Sum=i
    s=str(i)
    for j in s:
        Sum+=int(j)
    if Sum==N:
        return i
    return 0

t=N-len(str(N))*9 if N>18 else 1

if N==1:
    result=0
    
for i in range(t, N):
    result=Sep(i)
    if result!=0:
        break

    
print(result)